django_extensions/management/shells.py

```diff
@@ -34,6 +36,29 @@ class ObjectImportError(Exception):
     pass
 
 
+def get_app_name(mod_name):
+    """
+    Retrieves application name from models.py module path
+
+    >>> get_app_name('testapp.models.foo')
+    'testapp'
+
+    'testapp' instead of 'some.testapp' for compatibility:
+    >>> get_app_name('some.testapp.models.foo')
+    'testapp'
+    """
+    parts = mod_name.split('.')
+    try:
+        try:
+            return parts[parts.index(MODELS_MODULE_NAME) - 1]
```

Comment:
Should we use:
```python
rparts = list(reversed(parts))
return rparts[rparts.index(MODELS_MODULE_NAME) + 1]
```

for something like:
```python
>>> get_app_name('some.models.testapp.models.foo')
'testapp'
```

Avoiding getting `some` as a result there.

Reply:
good catch